src/app/Dashboard/DraggableRef.tsx

```diff
@@ -35,125 +35,327 @@
+                if (!di.isDraggingHost) {
+                  resetDroppableItem(di);
+                }
+              }
+            } else {
+              // reset when the mouse is not hovering on a card or between two cards
+              if (!di.isDraggingHost && hoveringDroppable.current == null) {
+                resetDroppableItem(di);
+              }
+            }
+          }
+        });
+      }
+      setRefStyle({
+        ...refStyle,
+        position: 'relative',
+        zIndex: 5000,
+        transition: '',
+        transform: `translate(${ev.pageX - startCoords.current[0]}px, ${ev.pageY - startCoords.current[1]}px)`,
+      });
     },
-    [callbackMouseMove, callbackMouseUp]
+    [setRefStyle, refStyle]
   );
 
+  const onDragStart = React.useCallback(
+    (ev: React.DragEvent<HTMLDivElement>) => {
+      ev.preventDefault();
+      if (isDragging) {
+        return;
+      }
+      const dragging = ev.target as HTMLElement;
+      const rect = dragging.getBoundingClientRect();
+
+      const draggableNodes: HTMLElement[] = Array.from(document.querySelectorAll(`div.${draggableRefKlazz}-wrapper`));
+      const droppableItems: DroppableItem[] = draggableNodes.reduce((acc: DroppableItem[], cur) => {
+        const isDraggingHost = cur.contains(dragging);
+        const droppableItem: DroppableItem = {
+          index: getOrder(cur),
+          node: cur,
+          rect: cur.getBoundingClientRect(),
+          isDraggingHost: isDraggingHost,
+          isHovered: false,
+        };
+        acc.push(droppableItem);
+        return acc;
+      }, []);
+
+      const initStyle = {
+        ...refStyle,
+        position: 'relative',
+        top: rect.y,
+        left: rect.x,
+        width: rect.width,
+        height: rect.height,
+        zIndex: 5000,
+      };
+
+      startCoords.current = [ev.pageX, ev.pageY];
+      setRefStyle(initStyle);
+      setIsDragging(true);
+      mouseMoveListener.current = (ev) => onMouseMoveWhileDragging(ev as MouseEvent, droppableItems);
+      mouseUpListener.current = () => onMouseUpWhileDragging(droppableItems);
+      document.addEventListener('mousemove', mouseMoveListener.current);
+      document.addEventListener('mouseup', mouseUpListener.current);
+    },
+    [setRefStyle, setIsDragging, onMouseMoveWhileDragging, onMouseUpWhileDragging, refStyle, isDragging]
+  );
+
+  const variableAttribute = React.useMemo(() => {
+    return { [dashboardCardOrderAttribute]: dashboardId };
+  }, [dashboardId]);
+
   return (
     <div
-      onMouseDown={handleOnMouseDown}
-      style={{
-        cursor: 'col-resize',
-        position: 'relative',
-        borderRight: '4px outset black',
-        borderRadius: '16px',
-      }}
-    />
+      ref={wrapperRef}
+      className={css(`${draggableRefKlazz}-wrapper`)}
+      onDragStart={onDragStart}
+      onTransitionEnd={onTransitionEnd}
+      style={{ ...refStyle }}
+      {...variableAttribute}
```

Comment:
Its best to use `data-*` attribute here https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes

This way, we can pull from `Element.dataset` consistently?